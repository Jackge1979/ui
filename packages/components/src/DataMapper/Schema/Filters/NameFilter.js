import Filter from './Filter';

export const ID = 'name-filter';

export default class NameFilter extends Filter {
	constructor(active) {
		super(ID, active);
	}

	setName(name) {
		this.name = name;
	}

	getName() {
		return name;
	}

	select(dataAccessor, schema, element) {
		return dataAccessor.getElementName(element).match(this.name);
	}
}
