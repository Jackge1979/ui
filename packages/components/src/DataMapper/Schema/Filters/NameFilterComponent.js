import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FilterBar from '../../../FilterBar';

export default class NameFilterComponent extends Component {

  constructor(props) {
    super(props);
    this.state = { docked: true };
    this.onToggle = this.onToggle.bind(this);
    this.onFilter = this.onFilter.bind(this);
  }

  onToggle() {
    this.props.filter.setActive(false);
    this.props.filter.setName(null);
    this.setState(prevState => ({
      docked: !prevState.docked,
    }));
    if (!this.state.docked) {
      this.props.onFilterChange(this.props.filter);
    }
  }

  onFilter(ev, value) {
    this.props.filter.setActive(value);
    this.props.filter.setName(value);
    this.props.onFilterChange(this.props.filter);
  }

  render() {
    return (
      <FilterBar
        id="filter-schema"
        className="filter"
        placeholder="Filter..."
        dockable
        docked={this.state.docked}
        navbar
        onToggle={this.onToggle}
        onFilter={this.onFilter}
      />
    );
  }
}

NameFilterComponent.propTypes = {
	filter: PropTypes.object,
	onFilterChange: PropTypes.func,
};
