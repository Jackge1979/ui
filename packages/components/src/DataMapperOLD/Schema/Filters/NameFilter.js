import Filter from './Filter';

export const ID = 'name-filter';

/**
 * A filter based on a name.
 */
export default class NameFilter extends Filter {
	constructor(active) {
		super(ID, active);
	}

	setName(name) {
		this.name = name;
	}

	getName() {
		return name;
	}

	select(dataAccessor, schema, element) {
		const elemName = dataAccessor.getElementName(element);
		return elemName.match(this.name) || elemName.toLowerCase().match(this.name.toLowerCase());
	}
}
