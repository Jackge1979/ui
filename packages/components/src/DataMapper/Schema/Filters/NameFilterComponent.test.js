//import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
//import TestUtils from 'react-dom/test-utils';
import NameFilterComponent from './NameFilterComponent';
import NameFilter from './NameFilter';
// import * as TestData from './TestData';
// import DefaultDataAccessor from '../../DefaultDataAccessor';
// import DataAccessorWrapper from '../../DataAccessorWrapper';

//const dataAccessor = new DataAccessorWrapper(new DefaultDataAccessor());

it('name-filter-component', () => {
  const filter = new NameFilter(false);
	// create React tree
	const tree = renderer
		.create(
			<NameFilterComponent
				filter={filter}
			/>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

// it('name-filter-component activate', () => {
//
//   const filter = new NameFilter(false);
//   const onFilterChange = jest.fn();
//
//   const filterComp = (
//     <NameFilterComponent
//       filter={filter}
//       onFilterChange={onFilterChange}
//     />
//   );
//   const wrapper = mount(filterComp);
//   wrapper
//     .find('.filter-name')
//     .at(0)
//     .simulate('click');
//
//
//   //expect(clearMapping).toBeCalled();
// });
