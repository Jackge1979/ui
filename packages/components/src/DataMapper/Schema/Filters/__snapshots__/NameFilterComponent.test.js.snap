// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`name-filter-component 1`] = `
<button
  aria-label="Toggle filter"
  className="filter-name btn btn-link"
  disabled={false}
  id="filter-schema"
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseOver={[Function]}
  role={null}
  type="button"
>
  <svg
    aria-hidden="true"
    className="theme-tc-svg-icon tc-svg-icon"
    focusable="false"
    title={null}
  >
    <use
      xlinkHref="#talend-search"
    />
  </svg>
</button>
`;
